vpr/src/route/rr_graph_area.cpp

```diff
@@ -226,24 +224,24 @@ void count_bidir_routing_transistors(int num_switch, int wire_to_ipin_switch, fl
                 /* Now add in the shared buffer transistors, and reset some flags. */
 
                 if (from_rr_type == CHANX) {
-                    for (i = device_ctx.rr_nodes[from_node].xlow() - 1;
-                         i <= device_ctx.rr_nodes[from_node].xhigh(); i++) {
+                    for (i = rr_graph.node_xlow(RRNodeId(from_node)) - 1;
```

Comment:
Changing the declared type of from_node and to_node to RRNodeId may let you get rid of some casting.